examples/src/main/java/com/squareup/moshi/recipes/Unwrap.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi.recipes;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.JsonQualifier;
+import com.squareup.moshi.JsonReader;
+import com.squareup.moshi.JsonWriter;
+import com.squareup.moshi.Moshi;
+import com.squareup.moshi.Types;
+import com.squareup.moshi.recipes.Unwrap.EnvelopJsonAdapter.Enveloped;
+import com.squareup.moshi.recipes.models.Card;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Retention;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Set;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+final class Unwrap {
+  @Enveloped private static Object unused;
```

Comment:
It’s probably the best we can do. I think we probably want a mechanism on Moshi where you can specify an annotation using a type literal `Enveloped.class` and just expand it out to the only reasonable implementation. I can do that, or you can read about dynamic proxies and annotation’s `hashCode()` method if you wanna try it.